dbms/programs/client/Client.cpp

```diff
@@ -658,6 +660,26 @@ class Client : public Poco::Util::Application
         return boost::replace_all_copy(prompt_by_server_display_name, "{database}", config().getString("database", "default"));
     }
 
+    /// Handle interrupt signal to clear current query by pressing Ctrl+C
+    /// or to exit if query is empty
+    static void signalHandler(int signo)
+    {
+        if (signo == SIGINT)
+        {
+            if (!rl_line_buffer[0])
+            {
+                std::cout << std::endl << "Bye." << std::endl;
+                exit(SIGINT);
```

Comment:
Destructor if Client class wont be invoked. ASan and Valgrind will report false positive memory leaks.